### DISCUSSION TITLE
How to change current directory of shell to selected directory in yazi upon exiting?

### DISCUSSION THREAD

**Discussion by fenneltheloon** [sxyazi/yazi#1034] [1. Q&A]
I've tried the function that is given on the website quick start:

```bash
function yy() {
	local tmp="$(mktemp -t "yazi-cwd.XXXXXX")"
	yazi "$@" --cwd-file="$tmp"
	if cwd="$(cat -- "$tmp")" && [ -n "$cwd" ] && [ "$cwd" != "$PWD" ]; then
		cd -- "$cwd"
	fi
	rm -f -- "$tmp"
}
```

but that doesn't seem to work when I use `q` or `Q` to exit the explorer. Is there something else that I should be doing to make this work?

**Comment by lpnh** [comment]
Hm, it should just work. Unfortunately, the suggestions I can think of are very basic. Like, making sure you:

- get the correct wrapper for your shell
- add it to your shell configuration file
- call `yy` instead of `yazi`
- enter the directory instead of opening it or just highlighting it

edit: improve the clarity

**Comment by erlangparasu** [comment]
i am using zsh (in `~/.zshrc` file)
i add these lines:
```shell

function yy() {
	local tmp="$(mktemp -t "yazi-cwd.XXXXXX")"
	yazi "$@" --cwd-file="$tmp"
	if cwd="$(cat -- "$tmp")" && [ -n "$cwd" ] && [ "$cwd" != "$PWD" ]; then
		cd -- "$cwd"
	fi
	rm -f -- "$tmp"
}

```

Calling `yy` on terminal, then navigate to some dir, then exit using `q`, it works.
But, using `Q`, not working.

**Reply by sxyazi** [reply]
> But, using `Q`, not working.

This is the expected behavior, the `Q` is designed for the purpose of not changing the CWD on quitting Yazi

**Reply by erlangparasu** [reply]
oh thanks

**Reply by Charari-2000** [reply]
Thank you so much! 
It works on me.
Just don’t forget to update the prompt, and it’ll be perfect!
```zsh
function yy() {
	local tmp="$(mktemp -t "yazi-cwd.XXXXXX")"
	yazi "$@" --cwd-file="$tmp"
	if cwd="$(cat -- "$tmp")" && [ -n "$cwd" ] && [ "$cwd" != "$PWD" ]; then
		cd -- "$cwd"
                zle reset-prompt
	fi
	rm -f -- "$tmp"
}
```

**Comment by AnirudhG07** [comment]
It doesnt work for me neither. I have copied the same from Quickstart and added to my zshrc

**Comment by kqvanity** [comment]
@sxyazi 
Is there a more self-contained way to switch to a directory instead of this one? ranger for instance allows mapping `S` to exit and switch the CWD to the highlighted directory

**Comment by rijkvp** [comment]
If anyone's looking for a version that works in fish:
```fish
function yy
    set -l tmp (mktemp -t "yazi-cwd.XXXXXX")
    yazi $argv --cwd-file=$tmp
    set cwd (cat -- $tmp)
    if test -s $tmp
        if test -n "$cwd" -a "$cwd" != "$PWD"
            cd -- $cwd
            commandline -f repaint # updates the prompt
        end
    end
    rm -f -- $tmp
end
```
Edit: added line to update the prompt

**Reply by lpnh** [reply]
Thanks for sharing! Just a quick heads up for anyone who might come across this discussion in the future: the documentation includes a version of the wrapper for most available shells. You can check it out here: https://yazi-rs.github.io/docs/quick-start#shell-wrapper
